### PR TITLE
parse_yaml.py: keep the trailing newline

### PR DESCRIPTION
**Pull request by BrunoB81HK**
Hi!

I have been using your library for a while and it's really great!

I have a very particular issue that is probably very niche, but I decided to try and fix it. As you can see in the pull request, the fix is very simple.

My use case is very particular, I use docker to prebuild my workspace and then I hop into the container to develop. When prebuilding with `--symlink-install`, this package create the module inside my `src/` directory. 

The problem arise when I mount my workspace inside the container. It essentially wipe the generated modules. The solution I use is to rebuild the packages inside my container and then commit the generated modules.

Which comes to the main point of this pull request, the trailing new line. Since most linter and formatter enforce a newline at the end of a python file by default, the fact that the generated modules don't have one is a little bit annoying.

tl;dr: For the (few) packages that somehow commit the generated modules, adding a trailing newline to the generated module could be a nice quality of life upgrade.

### CODE REVIEW

**Comment by pac48** [comment]
Can you run pre-commit to format your change `pre-commit run -a`. Otherwise, it looks good.

**Comment by BrunoB81HK** [comment]
@pac48 I'm so used to it being install on my repo that I forgot to install the hook before committing. This is done!

**Comment by pac48** [comment]
@BrunoB81HK Looks good now. I will merge it if you can just merge your changes with main.

**Comment by BrunoB81HK** [comment]
@pac48 I think its done

**Comment by tylerjw** [comment]
pre-commit is still failing with a formatting issue, do you mind running that and pushing the change?

**Comment by BrunoB81HK** [comment]
@tylerjw I don't mind. I will do it tomorrow.

**Comment by BrunoB81HK** [comment]
@pac48 @tylerjw Since the `main` branch was updated and it included changes to the `.pre-commit-config.yaml` file, the PR's commits were all over the place. I rebased the branch onto the latest commit on `main`, ran `pre-commit` and squashed all my commits into one. It should now be resolved.

**Comment by tylerjw** [comment]
I'm sorry for the difficulty getting this merged. I just kicked off ci again and will merge when it passes.

**Review by tylerjw — APPROVED** [review]

**Comment by BrunoB81HK** [comment]
Great! Thanks a lot guys for your time!
